base/compiler/ssair/inlining.jl

```diff
@@ -299,6 +299,15 @@ function finish_cfg_inline!(state::CFGInliningState)
     end
 end
 
+function transform_opaque_env!(ir::IRCode, @nospecialize(stmt′))
+    # If this is an access into the opaque closure environment,
+    # rewrite it to a call to `getfield_opaque_env`
+    if isexpr(stmt′, :call) && is_known_call(stmt′, getfield, ir, Any[]) && stmt′.args[2] === Argument(1)
+        stmt′.args[1] = Core.getfield_opaque_env
+    end
```

Comment:
This doesn't feel like the right transform, though I see why it should mostly work for the IR generated in the frontend. I think this should instead be a transform on the `argexpr[1]` of the caller from `OpaqueClosure` -> `insert_node!(:(getfield(OpaqueClosure, :env)))`

Thoughts?

Reply:
Yeah, that might be better. The reason I didn't do that is that I don't think the current setup with a separate object for the captures is what I want to do long term, but if and when we do that we'll probably have to revisit this anyway.